conjure-go-client/httpclient/internal/request_retrier.go

```diff
@@ -65,119 +55,73 @@ func (r *RequestRetrier) attemptsRemaining() bool {
 	return r.attemptCount < r.maxAttempts
 }
 
-// GetNextURI returns the next URI a client should use, or empty string if no suitable URI remaining to retry.
-// isRelocated is true when the URI comes from a redirect's Location header. In this case, it already includes the request path.
-func (r *RequestRetrier) GetNextURI(resp *http.Response, respErr error) (uri string, isRelocated bool) {
-	defer func() {
-		r.attemptCount++
-	}()
+// Next returns true if a subsequent request attempt should be attempted. If uses the previous response/resp err (if
+// provided) to determine if the request should be attempted. If the returned value is true, the retrier will have
+// waited the desired backoff interval before returning when applicable. If the previous response was a redirect, the
+// retrier will also return the URL that should be used for the new next request.
+func (r *RequestRetrier) Next(resp *http.Response, err error) (bool, *url.URL) {
```

Comment:
It feels weird to me that the retrier is still responsible for URI selection. It would read more clearly to me if the URI redirection logic was implemented behind the URI interfaces used in the client instead.